p2p/simulations/adapters/inproc.go

```diff
@@ -314,3 +347,107 @@ func (sn *SimNode) NodeInfo() *p2p.NodeInfo {
 	}
 	return server.NodeInfo()
 }
+
+// socketPipe creates an in process full duplex pipe based on OS sockets
+// credit to @lmars & Flynn
+// https://github.com/flynn/flynn/blob/master/host/containerinit/init.go#L743-L749
+// using this in large simulations requires raising OS's max open file limit
+func socketPipe() (net.Conn, net.Conn, error) {
+	pair, err := syscall.Socketpair(syscall.AF_UNIX, syscall.SOCK_STREAM, 0)
+	if err != nil {
+		return nil, nil, err
+	}
+	nameb := make([]byte, 8)
+	_, err = rand.Read(nameb)
+	if err != nil {
+		return nil, nil, err
+	}
+	f1 := os.NewFile(uintptr(pair[0]), string(nameb)+".out")
+	f2 := os.NewFile(uintptr(pair[1]), string(nameb)+".in")
+	pipe1, err := net.FileConn(f1)
+	if err != nil {
+		return nil, nil, err
+	}
+	pipe2, err := net.FileConn(f2)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return pipe1, pipe2, nil
+}
+
+func setSocketBuffer(conn net.Conn, socketReadBuffer int, socketWriteBuffer int) error {
+	switch v := conn.(type) {
+	case *net.UnixConn:
+		err := v.SetReadBuffer(socketReadBuffer)
+		if err != nil {
+			return err
+		}
+		err = v.SetWriteBuffer(socketWriteBuffer)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// netPipe wraps net.Pipe in a signature returning  an error
+func netPipe() (net.Conn, net.Conn, error) {
+	p1, p2 := net.Pipe()
+	return p1, p2, nil
+}
+
+// tcpPipe creates an in process full duplex pipe based on a localhost TCP socket
+func tcpPipe() (net.Conn, net.Conn, error) {
```

Comment:
This must go away.

Reply:
@nonsense `go away` as in `go away with all of the code which depends on it`? there's dependant code in `pss_test` and `snapshot_sync_test`

Reply:
We already have this in `rlpx_test.go` - we should export `TCPPipe` and reuse in this package.